### PR TITLE
Wait for logs in blockchain publisher

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- Gets the full logs of the transaction when submitting to the blockchain, so we can construct a valid `OriginatorEnvelope` that includes the `sequence_id`

## Notes
Once this is in I will update https://github.com/xmtp/xmtpd/pull/248 to take advantage of it

## Release Notes

- **New Features**
	- Enhanced blockchain message publishing with improved error handling and structured responses.
	- Introduced new methods for parsing transaction receipts related to message sending and identity updates.
	- Added a utility function to create a cancelled context for testing purposes.

- **Bug Fixes**
	- Refined error handling in blockchain publishing methods to provide clearer error messages and logging.

- **Tests**
	- Added unit tests for `BlockchainPublisher` methods to ensure expected functionality and error handling.

- **Documentation**
	- Updated method signatures to reflect new return types for blockchain publishing operations.

### CODE REVIEW

**Comment by neekolas** [comment]
* **#249** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/249?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pkg/blockchain/blockchainPublisher.go [hunk 1]
```diff
@@ -62,46 +64,92 @@ func (m *BlockchainPublisher) PublishGroupMessage(
...
+	errorMsg string,
+) (*T, error) {
+	for _, logEntry := range receipt.Logs {
+		if logEntry == nil {
+			continue
+		}
+		event, err := parse(*logEntry)
+		if err != nil {
+			continue
+		}
```
Comment (richardhuaaa) [inline pkg/blockchain/blockchainPublisher.go:150]: Can there be multiple logs for the same update type, e.g. submitted by others?

Comment (neekolas) [reply pkg/blockchain/blockchainPublisher.go:150]: The receipt is scoped to a single transaction, so it will only be your own events.
